### ISSUE TITLE
Key attribute for react

### ISSUE THREAD

**Issue by olegdunkan** [facebook/relay#1140]
Go to http://facebook.github.io/relay/ press F12, you will see

playground.js:7785 Warning: Each child in an array or iterator should have a unique "key" prop. Check the render method of `[object Object]`. See https://fb.me/react-warning-keys for more information.

There is no "key" attribute in next snippet:

    class TeaStore extends React.Component {
       render() { 
        return <ul>
           {this.props.store.teas.map(tea => <Tea tea={tea} />)}
        </ul>;
      }
    }
React says 

> The key should always be supplied directly to the components in the array, not to the container HTML child of each component in the array:
It had to be something like this:

    class TeaStore extends React.Component {
       render() { 
        return <ul>
           {this.props.store.teas.map(tea => <Tea key={tea.id} tea={tea} />)}
        </ul>;
      }
    }

Next question, can I use __dataId__ for these purposes?, but I think it not good idea. 
How can I prefetch id in fragment?

Let say I have
 
    fragment on Store
      teas { ${Tea.getFragment('tea')} },
    }

but I need something like these

    fragment on Store
      teas { 
                id,
                ${Tea.getFragment('tea')} },
    }

What is best method to suit the case?

**Comment by josephsavona** [comment]
I would recommend fetching `id` in the parent component and using the id as the key:

```graphql
fragment on Store
  teas { 
    id # fetch id here
    ${Tea.getFragment('tea')} 
  }
}
```

**Comment by olegdunkan** [comment]
Ok, thanks!

**Comment by holycrapbatmanyourright** [comment]
Unsubscribe please!

On Saturday, May 14, 2016, olegdunkan <notifications@github.com> wrote:

> Closed #1140 <https://github.com/facebook/relay/issues/1140>.
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/facebook/relay/issues/1140#event-660660063>
>


-- 
Sent from Gmail Mobile

**Comment by wincent** [comment]
@holycrapbatmanyourright: Click the "Unsubscribe" button in the right-hand side of the UI on [the issue's page](https://github.com/facebook/relay/issues/1140). (It's on the right-hand side in the web UI on desktop; not sure about mobile.)
